File: timecapsule-database/src/main/java/com/timecapsule/infra/config/DatabaseProperties.java
package com.timecapsule.infra.config;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;

@Getter
@RequiredArgsConstructor
@ConstructorBinding
@ConfigurationProperties(prefix = "spring.datasource")
public class DatabaseProperties {
    private final String username;
    private final String password;
    private final String driverClassName;
    private final String jpaHibernateDdlAuto;
    private final String url;
    private final Integer maximumPoolSize;
    private final Long connectionTimeout;
}

File: common/timecapsule-database/src/main/kotlin/com/timecapsule/database/config/DatabaseProperties.kt
package com.timecapsule.database.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "spring.datasource")
data class DatabaseProperties (
    val username: String,
    val password: String,
    val driverClassName: String,
    val jpaHibernateDdlAuto: String,
    val url: String,
    val maximumPoolSize: Int,
    val connectionTimeout: Long,
)
